### PR TITLE
[MOBILEAPPS-1622] Code that will check for the mobile app on the device(iOS and Android) for the URL and would redirect the user to the corresponding app if it is installed otherwise would open the link on the browser only.

### PR DESCRIPTION
**Pull request by jatin2008**
**Please check if the PR fulfills these requirements**

> - [x ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
same as working earlier

**What is the new behaviour?**
Code that will check for the mobile app on the device(iOS and Android) for the URL and would redirect the user to the corresponding app if it is installed otherwise would open the link on the browser only.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export class AcaMobileAppSwitcherService {
+  constructor(private config: AppConfigService) {}
+
+  checkForMobileApp() {
+    const mobileAppSwitchConfig: any = this.config.get<any>('mobileAppSwitch', {});
+    const ua = navigator.userAgent.toLowerCase();
+    const isAndroid = ua.indexOf('android') > -1;
+    const isIphone = ua.indexOf('iphone') > -1;
+    if (isIphone === true) {
+      const url = window.location.href;
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts:16]: you can move this outside the "if" as it's the same for "else" too

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        launchApp: () => {
+          window.location.href = mobileAppSwitchConfig.isIphone + url;
+        },
+        openWebApp: () => {
+          window.location.href = mobileAppSwitchConfig.appleStore;
+        }
+      };
+      app.launchApp();
+    } else if (isAndroid === true) {
+      const url = window.location.href;
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts:27]: same variable as in the "if", so can be made a const after `isIphone`

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        launchApp: () => {
+          setTimeout(() => {
+            window.location.href = mobileAppSwitchConfig.isAndroidPart1 + url + mobileAppSwitchConfig.isAndroidPart2;
+          }, 500);
+        },
+        openWebApp: () => {
+          window.location.href = mobileAppSwitchConfig.googlePlayStore;
+        }
+      };
+      app.launchApp();
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts:38]: do you need creating an "app" object just to call "app.launchApp()"? why now running that directly?

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { AppConfigService } from '@alfresco/adf-core';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AcaMobileAppSwitcherService {
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts:7]: missing unit tests

File: app/src/app.config.json.tpl [hunk 5]
```diff
@@ -6,6 +6,14 @@
   "providers": "${APP_CONFIG_PROVIDER}",
   "authType": "${APP_CONFIG_AUTH_TYPE}",
   "loginRoute": "login",
+  "mobileAppSwitch": {
+      "enabled" : "${APP_CONFIG_ENABLE_MOBILE_APP_SWITCH}",
+      "isIphone": "iosamw://",
```
Comment (DenysVuika) [inline app/src/app.config.json.tpl:11]: how this is going to be configured externally? should it have separate ENV variables for that?

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts (hunk 1 above)
Comment (jatin2008) [reply projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts]: noted and resolved

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts (hunk 2 above)
Comment (jatin2008) [reply projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts]: noted and resolved.

File: app/src/app.config.json.tpl (hunk 5 above)
Comment (jatin2008) [reply app/src/app.config.json.tpl:11]: yess I have configured seperate env variable as APP_CONFIG_ENABLE_MOBILE_APP_SWITCH in .env file, is that fine?

Comment (jatin2008) [reply app/src/app.config.json.tpl:11]: yess I have configured seperate env variable as APP_CONFIG_ENABLE_MOBILE_APP_SWITCH in .env file, is that fine?

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts (hunk 4 above)
Comment (jatin2008) [reply projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts:32]: unit test cases have been added

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        enabled: true,
+        isIphone: 'iosamw://',
+        isAndroidPart1: 'intent:///',
+        isAndroidPart2: '#Intent;scheme=androidamw;package=com.alfresco.content.app.debug;end'
+      }
+    });
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts:54]: this test can be removed as it does not make sense, it is already tested with other methods so we know it was created

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  });
+  describe('checkForMobileApp', () => {
+    it('should check if the indexof userAgent is `iphone`', () => {
+      const iphone = true;
+      service.checkForMobileApp();
+      expect(iphone).toBe(true);
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts:60]: expect `true` to be `true` ?

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    });
+    it('should check if the indexof userAgent is `android`', () => {
+      const android = true;
+      service.checkForMobileApp();
+      expect(android).toBe(true);
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts:65]: expect `true` to be `true` ?

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts (hunk 6 above)
Comment (jatin2008) [reply projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts:54]: noted and removed

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts (hunk 7 above)
Comment (jatin2008) [reply projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts:60]: noted and working on test cases for resolution

File: projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts (hunk 8 above)
Comment (jatin2008) [reply projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.spec.ts:65]: noted and working on test cases for resolution

**Comment by jatin2008** [comment]
Refer PR - https://github.com/Alfresco/alfresco-content-app/pull/2889

**Pull request closed without merging**